### DISCUSSION TITLE
Shulkers, Levitation, and End Cities?

### DISCUSSION THREAD

**Discussion by MagmaSlime123** [Roadhog360/Et-Futurum-Requiem#57] [Ideas]
Meow.

**Comment by Roadhog360** [comment]
Yes, this is something I've gotten my mind on for a while. Even during my off time I'm still thinking of how to approach it. As you may know my skill in coding is not very high so I might try to get my other friends (jss, Greg) to help me out with these features.

**Comment by GregoriusT** [comment]
I presume you mean that Potion Effect? An Idea would be to NOT use the Potion Effect List and just add the Levitation Effect with some NBT Flag in the Entity that got hit by Shulkers.

**Reply by Roadhog360** [reply]
Both would be an option. How would I actually go about extending the NBT of a vanilla entity? Don't you add NBT by implementing it in an entity class?

Also, the potion list can easily be  extended. Thaumcraft does it for its own potions, so I'll do it too.

**Reply by GregoriusT** [reply]
Oh it is as easy as "entity.getEntityData().setBoolean("yourvalue", true);", whatever values you set to the NBTTagCompound regardless of Boolean, Integer, String etc. Will stay saved in that Entity forever. This way you can store values such as "floatiness" whenever a Shulker hits you, without having to resort to Potion Effects.

**Reply by Roadhog360** [reply]
That's kickass. Thanks man.

**Reply by MagmaSlime123** [reply]
So, issue resolved I guess?

**Reply by Roadhog360** [reply]
> Oh it is as easy as "entity.getEntityData().setBoolean("yourvalue", true);", whatever values you set to the NBTTagCompound regardless of Boolean, Integer, String etc. Will stay saved in that Entity forever. This way you can store values such as "floatiness" whenever a Shulker hits you, without having to resort to Potion Effects.

Due to needing to check this every tick, using NBT would be very slow, so a potion ID will need to be used. There are mods which can safely expand the potion ID limit should you need to.

**Comment by Roadhog360** [comment]
End cities will be attempted in 2.6.0. Closing preemptively as every other feature is there and end cities *will* be attempted for 2.6.0.
